python/pyarrow/_dataset.pyx

```diff
@@ -1321,38 +1321,36 @@ cdef class DirectoryPartitioning(Partitioning):
         self.directory_partitioning = <CDirectoryPartitioning*> sp.get()
 
     @staticmethod
-    def discover(field_names, object max_partition_dictionary_size=0):
+    def discover(field_names, inspect_dictionary=False):
```

Comment:
It might be useful to still accept `max_partition_dictionary_size=None/-1` for now, and if that is passed, set `inspect_dictionary` to True. 
That would help a bit for dask to write a parquet reader that works both on pyarrow 1.0 and 2.0, on the short term.

Reply:
Alright, I'll add that keyword back and document that it has been replaced